core/src/main/java/com/graphhopper/routing/AlternativeRouteCH.java

```diff
@@ -191,6 +195,17 @@ public int compare(AlternativeInfo o1, AlternativeInfo o2) {
         return alternatives;
     }
 
+    private List<SPTEntry> calcSPEList(final SPTEntry fromSPTEntry, final SPTEntry toSPTEntry) {
+        final List<SPTEntry> list = new ArrayList<>();
+        new NodeBasedCHBidirPathExtractor(graph, graph.getBaseGraph(), weighting) {
```

Comment:
Since we are not even unpacking any shortcuts here using BidirPathExtractor (without CH) should be enough ? To further clean this up we could create something that is *only* concerned with following the parent pointers (and provides some hooks like onEdge etc.) (and not with `Path` at all) and *use* this within the path extractor. But not sure if this is worth the effort atm.